lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -265,16 +274,30 @@ Function Compare-PropertyList {
         if ($existing_property -eq $null) {
             # we have more properties than before,just add to the new
             # properties list
-            $diff_list = [System.Collections.ArrayList]@()
-            foreach ($property_arg in $total_args) {
-                if ($new_property.ContainsKey($property_arg)) {
-                    $com_name = Convert-SnakeToPascalCase -snake $property_arg
-                    $property_value = $new_property.$property_arg
-                    [void]$diff_list.Add("+$com_name=$property_value")
+            $properties = $new_property.Keys | Where-Object { $total_args -contains $_ }
+            if ($properties -ne $null) {
+                $diff_list = [System.Collections.ArrayList]@()
+
+                foreach ($property_arg in $properties) {
+                    if ($new_property.ContainsKey($property_arg)) {
+                        $com_name = Convert-SnakeToPascalCase -snake $property_arg
+                        $property_value = $new_property.$property_arg
+
+                        if ($property_value -is [Hashtable]) {
+                            foreach ($sub_property_arg in $property_value.Keys) {
+                                $sub_com_name = Convert-SnakeToPascalCase -snake $sub_property_arg
+                                $sub_property_value = $property_value.$sub_property_arg
+                                [void]$diff_list.Add("+$sub_com_name=$sub_property_value")
```

Comment:
You probably want this to be `[void]$diff_list.Add("+$($com_name).$sub_com_name=$sub_property_value")` so we know it is a nested property of a trigger.